Merge to Dev: Combined Code from everyone

NMeyer8386 opened the pull request.

NMeyer8386 committed:
Import Class Crud (#7)

* User mgmt initial commit (#1) (#2)

* first commit

* made a test view + installed laravel-more-command

* Added 'theanik/laravel-more-command'

* created test controller and added a route for it

* fixed redis issues

* Created initial layout view of header, nav, footer, and created common css directory and file in public directory

* Turned header, nav, footer to base layout and added carousel component.

* stopped tracking env file

* initial commit 2

* Merge attempt setup

* Delete .env:Zone.Identifier

Uneeded file

* Delete app.dockerfile:Zone.Identifier

Uneeded file

* Delete docker-compose.yaml:Zone.Identifier

Uneeded file

* Delete env:Zone.Identifier

Un-Needed file

* Update package-lock.json

returned name to "open_road"

* Delete ppsBanner1.jpg:Zone.Identifier

Un-Needed file

* Delete ppsBanner2.jpg:Zone.Identifier

Un-Needed file

* Delete ppsBanner3.jpg:Zone.Identifier

Un-Needed file

* Delete redis.conf

Private file, environment specific

* Delete redis.conf:Zone.Identifier

Un-Needed file

* Delete users.acl

Private information, environment specific

* Delete users.acl:Zone.Identifier

Un-Needed file

* First Draft of Migrations

* Second Draft

* number 3

* added a reference to STUFF_ID on the users table

* Fixed migrations so they actually work :)

* Changed migrations so they work properly

* Update composer.json

Removed unexpected dependency

* Delete composer.lock

Removed from commit due to content added from unexpected dependency

* Update package-lock.json

* Delete redis.conf

Removed file

* Delete users.acl

removed file

* removed a thing that probably shouldn't be there in the first place

* Fixed Database

* did some stuff

---------

Co-authored-by: Streibel-LC <90427932+Streibel-LC@users.noreply.github.com>
Co-authored-by: Streibel-LC <timothy.streibel@lehtbridgecollege.ca>
Co-authored-by: JoshPHickman <josh.hickman@lethbridgecollege.ca>
Co-authored-by: “BryanTowpich” <bryan.towpich@lethbridgecollege.ca>
Co-authored-by: BryanTowpich <90427100+BryanTowpich@users.noreply.github.com>

* Created Admin Controller to handle admin operations and added content to the dashboard view

* Added a conditional statement to display admin dashboard when on a specific route

* Removed responses because those tended to break things

* Made course elements guarded and included soft deletes

* changed a route for future use

* added relevant routes and removed irrelevant ones

* deleted the test controller and page that are no longer needed

* re-committing composer.lock

* changed CLASS_ID to id to fix route-model binding

* Updated migrations

* Changing COURSE_ID to make route-model binding work

* moved blades from course_crud

* updated links to navigate to class functions

* Created necessary views for CRUD operation: still have to pass dropdowns to controller

* Created ClassController to handle relevant navigation and operations

* Updated CourseController to be on par with Course_crud branch for testing

* Added guarded and softdeletes

* Brought blades up to par with course_crud

* fixed the isset call for the carousel

* Added routing for ClassController and inported necessary classes

* Migration Fixes

* Comments for Code Review

---------

Co-authored-by: Streibel-LC <90427932+Streibel-LC@users.noreply.github.com>
Co-authored-by: Streibel-LC <timothy.streibel@lehtbridgecollege.ca>
Co-authored-by: JoshPHickman <josh.hickman@lethbridgecollege.ca>
Co-authored-by: “BryanTowpich” <bryan.towpich@lethbridgecollege.ca>
Co-authored-by: BryanTowpich <90427100+BryanTowpich@users.noreply.github.com>

NMeyer8386 committed:
Import User Crud (#9)

* first commit

* made a test view + installed laravel-more-command

* Added 'theanik/laravel-more-command'

* created test controller and added a route for it

* fixed redis issues

* Created initial layout view of header, nav, footer, and created common css directory and file in public directory

* Turned header, nav, footer to base layout and added carousel component.

* stopped tracking env file

* initial commit 2

* Merge attempt setup

* Delete .env:Zone.Identifier

Uneeded file

* Delete app.dockerfile:Zone.Identifier

Uneeded file

* Delete docker-compose.yaml:Zone.Identifier

Uneeded file

* Delete env:Zone.Identifier

Un-Needed file

* Update package-lock.json

returned name to "open_road"

* Delete ppsBanner1.jpg:Zone.Identifier

Un-Needed file

* Delete ppsBanner2.jpg:Zone.Identifier

Un-Needed file

* Delete ppsBanner3.jpg:Zone.Identifier

Un-Needed file

* Delete redis.conf

Private file, environment specific

* Delete redis.conf:Zone.Identifier

Un-Needed file

* Delete users.acl

Private information, environment specific

* Delete users.acl:Zone.Identifier

Un-Needed file

* First Draft of Migrations

* Second Draft

* number 3

* added a reference to STUFF_ID on the users table

* Fixed migrations so they actually work :)

* Changed migrations so they work properly

* Update composer.json

Removed unexpected dependency

* Delete composer.lock

Removed from commit due to content added from unexpected dependency

* Update package-lock.json

* Delete redis.conf

Removed file

* Delete users.acl

removed file

* removed a thing that probably shouldn't be there in the first place

* Fixed Database

* did some stuff

* Stuff Create Read

* Updated The Migrations with id fields on some

* Not Verbose

---------

Co-authored-by: Streibel-LC <timothy.streibel@lehtbridgecollege.ca>
Co-authored-by: JoshPHickman <josh.hickman@lethbridgecollege.ca>
Co-authored-by: Streibel-LC <90427932+Streibel-LC@users.noreply.github.com>
Co-authored-by: “BryanTowpich” <bryan.towpich@lethbridgecollege.ca>
Co-authored-by: BryanTowpich <90427100+BryanTowpich@users.noreply.github.com>

JoshPHickman committed:
Webcontent mgmt (#8)

* User mgmt initial commit (#1) (#3)

* first commit

* made a test view + installed laravel-more-command

* Added 'theanik/laravel-more-command'

* created test controller and added a route for it

* fixed redis issues

* Created initial layout view of header, nav, footer, and created common css directory and file in public directory

* Turned header, nav, footer to base layout and added carousel component.

* stopped tracking env file

* initial commit 2

* Merge attempt setup

* Delete .env:Zone.Identifier

Uneeded file

* Delete app.dockerfile:Zone.Identifier

Uneeded file

* Delete docker-compose.yaml:Zone.Identifier

Uneeded file

* Delete env:Zone.Identifier

Un-Needed file

* Update package-lock.json

returned name to "open_road"

* Delete ppsBanner1.jpg:Zone.Identifier

Un-Needed file

* Delete ppsBanner2.jpg:Zone.Identifier

Un-Needed file

* Delete ppsBanner3.jpg:Zone.Identifier

Un-Needed file

* Delete redis.conf

Private file, environment specific

* Delete redis.conf:Zone.Identifier

Un-Needed file

* Delete users.acl

Private information, environment specific

* Delete users.acl:Zone.Identifier

Un-Needed file

* First Draft of Migrations

* Second Draft

* number 3

* added a reference to STUFF_ID on the users table

* Fixed migrations so they actually work :)

* Changed migrations so they work properly

* Update composer.json

Removed unexpected dependency

* Delete composer.lock

Removed from commit due to content added from unexpected dependency

* Update package-lock.json

* Delete redis.conf

Removed file

* Delete users.acl

removed file

* removed a thing that probably shouldn't be there in the first place

* Fixed Database

* did some stuff

---------

Co-authored-by: Streibel-LC <90427932+Streibel-LC@users.noreply.github.com>
Co-authored-by: Streibel-LC <timothy.streibel@lehtbridgecollege.ca>
Co-authored-by: JoshPHickman <josh.hickman@lethbridgecollege.ca>
Co-authored-by: “BryanTowpich” <bryan.towpich@lethbridgecollege.ca>
Co-authored-by: BryanTowpich <90427100+BryanTowpich@users.noreply.github.com>

* Made FAQ page, semi-working edit FAQ page

---------

Co-authored-by: NMeyer8386 <90421439+NMeyer8386@users.noreply.github.com>
Co-authored-by: Streibel-LC <90427932+Streibel-LC@users.noreply.github.com>
Co-authored-by: Streibel-LC <timothy.streibel@lehtbridgecollege.ca>
Co-authored-by: “BryanTowpich” <bryan.towpich@lethbridgecollege.ca>
Co-authored-by: BryanTowpich <90427100+BryanTowpich@users.noreply.github.com>
Co-authored-by: ClairePoulsen <claire.e.poulsen@gmail.com>

Commit:
made home nav element link to homepage

JoshPHickman reviewed (commented):
Merging all working code into the main branch.

Commit:
imported stuffcontroller

Commit:
fixed migration as per Bryan's orders

Commit:
Linked User_Crud operations to admin dashboard tiles

Commit:
Changed requests so data is passed properly

Commit:
Tried some stuff that didn't work

Commit:
name changed for some reason

Commit:
Auto incremented end date which can easily be applied for course duration

Commit:
removed a class that wasn't needed

Commit:
pointed to the right table and tried to cast dates to a format

Commit:
tried some more stuff that didn't work

Commit:
Added Spatie/laravel-permission depend. to app.

Commit:
Deleted some test files I made before starting litenotes

BryanTowpich committed:
Changed user id to uuid and put correct
dependendencies into the user model

BryanTowpich committed:
keytype string added now it works so im commiting
before i break way more things

JoshPHickman committed:
Delete Precision_Riding_Academy_most_up_to_date.png:Zone.Identifier

These make GitHub unhappy, I'll be sure to stop including them in future commits.

JoshPHickman committed:
Updated app layout & route file.

Added updated UI components to the base layout file, as well as making minor adjustments to the route file to accommodate additions.

BryanTowpich committed:
Class Dropdowns Fixed

JoshPHickman committed:
Updated carousel component to be dynamic.

Added the advertisement resource controller and route with upload view. Carousel will now populate from data in advertisement table. Added link to ad upload in admin dashboard. Updated app service provider to make ads globally available. Minor update to ad model to align with database changes.

JoshPHickman committed:
Updated sign in page.

Initial sign in page styled with bootstrap.

clairefleckney committed:
Made some progress towards making FAQ fully functional

clairefleckney committed:
Merge branch 'test' of https://github.com/SIT-SOFTware/open-road into test

Commit:
added route to registration

Commit:
Merge branch 'test' of https://github.com/SIT-SOFTware/open-road into test

Commit:
added truncation after date so time isn't included

JoshPHickman committed:
Updated advertisement components to align with new db migrations.

Updated ad model and controller to align with new column names, while accepting auto incremented values. Updated migrations to auto increment primary id. Added conditional logic for checking if slides exist before rendering.

JoshPHickman committed:
Merge branch 'test' of https://github.com/SIT-SOFTware/open-road into test

Commit:
Merge branch 'test' of https://github.com/SIT-SOFTware/open-road into test

Commit:
changed which field is saved to the db since id is auto created

Commit:
Changed course id so it displays the course code and not the uuid

Commit:
changed to show course code instead of course uuid
